CONTRIBUTING.md

```diff
@@ -1,3 +1,34 @@
+# Contributing to Elasticsearch Ruby
+
+To work on the code, clone and bootstrap the project first:
+
+```
+git clone https://github.com/elasticsearch/elasticsearch-ruby.git
+cd elasticsearch-ruby/
+bundle exec rake bundle
+```
+
+This will run `bundle install` in all subprojects. 
+
+To run the tests, you need to start a testing cluster on port 9250, or provide a different one in the `TEST_CLUSTER_PORT` environment variable.
+
+There's a Rake task to start a testing cluster in a Docker container:
+
+`rake docker:start[version]` - E.g.: `rake docker:start[7.x-SNAPSHOT]`. To start the container with Platinum, pass it in as a parameter: `rake docker:start[7.x-SNAPSHOT,xpack]`.
+
+
+To run all the tests in all the subprojects, use the Rake task:
+
+```
+time rake test:client
+```
+
+By default, tests will atempt to use `http://localhost:9200` as a test server. If you're using a different host/port, set the `TEST_ES_SERVER` environment variable, e.g.:
+
+```
+$ TEST_ES_SERVER='http://localhost:9250' be rake test:client
+```
+
 The process for contributing to any of the [Elasticsearch](https://github.com/elasticsearch) repositories is similar:
 
 1. It is best to do your work in a separate Git branch. This makes it easier to synchronise your changes with [`rebase`](http://mislav.uniqpath.com/2013/02/merge-vs-rebase/).
```

Comment:
Could also add a note that all submissions should include tests, and that external contributions require signing the CLA.

Reply:
Yeah, these points are specified in the second and third items.

Reply:
Ah yes, sorry should have expanded downwards!